app/components/collections/detail_component.html.erb

```diff
@@ -1,7 +1,7 @@
 <table class="table table-sm mb-5">
   <thead class="table-light">
     <tr>
-      <th scope="col" class="table-title col-3">Details</th>
+      <th scope="col" class="table-title col-3">Details<%= edit_link('#', "Edit details") %></th>
```

Comment:
Where is this edit_link method defined?

Reply:
https://github.com/sul-dlss/happy-heron/pull/874/files#diff-54534c85a1e46345b7a8c23b23c570bb737ed9ad39c06a725b86a7b08cd98d1cR16 

And I suspect that's why all the inheritance was added.

Reply:
Exactly correct.  Could use a mixin if we don't like the inheritance?  All of those components with inheritance are related to the collection detail page, which was the logic in doing it that way.

This method of defining an "edit_link" method is also used in the Work detail page, but in that case there is a one component for the whole page instead of breaking it out into sections like we do for the collection, so no need to for inheritance there to avoid duplication.

Reply:
Maybe define `edit_link` as a helper method in `ApplicationController`, like `#user_with_groups`, and invoke it in components that need it via `helper.edit_link` ?

Reply:
yeah, we could do that, though then I would argue we call it something like `collection_edit_link` and do the same with the work to make it all consistent (by moving it from the work component to the application controller)

Reply:
@peetucket dig that.

Reply:
Perhaps make EditLinkComponent

Reply:
and same for the Work for consistency?  in this PR or maybe follow on?

Reply:
Side note: this consistency with `Works` was completed in #891